core/src/main/python/update.py

```diff
@@ -125,9 +126,11 @@ def __update_online(model, model_context, aliases):
 
     try:
         __wlst_helper.connect(admin_user, admin_pwd, admin_url)
-        deployer_utils.ensure_no_uncommitted_changes_or_edit_sessions()
+        deployer_utils.ensure_no_uncommitted_changes_or_edit_sessions(model_context.is_undo_current_edit())
         __wlst_helper.edit()
         __wlst_helper.start_edit()
+        if model_context.is_undo_current_edit():
+            deployer_utils.undo_current_edit()
```

Comment:
Does undo_current_edit need to happen before the new edit is started?

Reply:
Yes, it needs to have an edit session first